### PR TITLE
Enable ErrorProne FallThrough, MissingCasesInEnumSwitch, OrphanedFormatString

### PR DESCRIPTION
**Pull request by ebyhr**
# Description

Enable ErrorProne FallThrough, MissingCasesInEnumSwitch, OrphanedFormatString

## Type of change

Please delete options that are not relevant.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code

### CODE REVIEW

File: polaris-service/src/main/java/io/polaris/service/catalog/PolarisCatalogHandlerWrapper.java [hunk 1]
```diff
@@ -578,7 +578,7 @@ public LoadTableResponse createTableDirectWithWriteDelegation(
                   .withLocation(request.location())
                   .withPartitionSpec(request.spec())
                   .withSortOrder(request.writeOrder())
-                  .withProperties(request.properties())
+                  .withProperties(properties)
```
Comment (snazy) [inline polaris-service/src/main/java/io/polaris/service/catalog/PolarisCatalogHandlerWrapper.java:581]: Looks like this change doesn't fit into the description/title of this PR. Mind moving this change to a separate PR?

Comment (ebyhr) [reply polaris-service/src/main/java/io/polaris/service/catalog/PolarisCatalogHandlerWrapper.java]: Separated into https://github.com/apache/polaris/pull/207

**Review by sfc-gh-emaynard — APPROVED** [review]

**Review by snazy — APPROVED** [review]
Thanks @ebyhr
